### ISSUE TITLE
Sacado:  `SacadoViewFill` should be used only if needed

### ISSUE THREAD

**Issue by romintomasetti** [trilinos/Trilinos#11873]
## Enhancement

It seems that `Kokkos::Impl::SacadoViewFill` for non-device views could just be:
```c++
template <class OutputView, typename Enabled = void>
struct SacadoViewFill : Kokkos::Impl::ViewFill<OutputView> {};
```

This would allow a nice reuse of Kokkos team's improvements for view fill.

**Comment by GrahamBenHarper** [comment]
@trilinos/sacado @etphipp

**Comment by etphipp** [comment]
Sounds reasonable.  Would you mind creating a PR?

**Comment by etphipp** [comment]
The reason for that is the Sacado HIP implementation doesn't yet support mapping vector parallelism across the derivative dimension, which is what `partition_scalar` is for.  Adding support for that in HIP is not challenging, but is a larger effort.

**Comment by masterleinad** [comment]
As soon as you rely on something in the `Kokkos::Impl` namespace that is not a customization point that you actually specialize, you are on your own and there is no guarantee for backward-compatible changes.

**Comment by romintomasetti** [comment]
@masterleinad I agree that `Kokkos::Impl` shouldn't be used by the "casual" `Kokkos` user. However, especially in this case, specializing the view of `Sacado` objects forces the "specializer" to re-write what `Kokkos::Impl::ViewFill` does even if the re-write could simply re-use `Kokkos::Impl::ViewFill`. I guess in that case it is OK to use `Kokkos::Impl`? If you feel we shouldn't use `Kokkos::Impl` for the "default" `SacadoViewFill`, what do you suggest then? :smiley:

**Comment by masterleinad** [comment]
> @masterleinad I agree that `Kokkos::Impl` shouldn't be used by the "casual" `Kokkos` user. However, especially in this case, specializing the view of `Sacado` objects forces the "specializer" to re-write what `Kokkos::Impl::ViewFill` does even if the re-write could simply re-use `Kokkos::Impl::ViewFill`. I guess in that case it is OK to use `Kokkos::Impl`? If you feel we shouldn't use `Kokkos::Impl` for the "default" `SacadoViewFill`, what do you suggest then? 😃

Sure, if `Stokhos` maintains that and doesn't blame `Kokkos` if this breaks because we change `Kokkos::Impl::ViewFill`.

**Comment by romintomasetti** [comment]
@masterleinad Should we reopen this one?

**Comment by masterleinad** [comment]
> @masterleinad Should we reopen this one?

I don't think I'm the right person to ask. 🙂

**Comment by romintomasetti** [comment]
@etphipp ?

**Comment by etphipp** [comment]
@romintomasetti were you going to work on a PR for this?

**Comment by romintomasetti** [comment]
Given the ongoing refactoring of `Kokkos::View` to leverage `std::mdspan`, I think this issue will need to wait for that to complete. However, it’s definitely worth keeping tabs on — once the `std::mdspan` migration is finished, the specialization will still need to be implemented in a way that avoids unnecessary duplication.

**Comment by masterleinad** [comment]
https://github.com/kokkos/kokkos/pull/8152 should make any extra `deep_copy` overloads for `Sacado` types unnecessary.
